### PR TITLE
Fix check for unrelated decks playing when starting Auto DJ

### LINKED ISSUE #13734: Auto DJ: Wrong error message about playing deck 3 and 4 if they are uses as Auto DJ decks
### Bug Description

The message: 
"Decks 3 and 4 must be stopped to enable Auto DJ mode."
is wrong if deck 3 and 4 are used for Auto DJ. 

### Version

2.4.1

### OS

Ubunut Focal

### PR DESCRIPTION
**Pull request by daschuer**
This fixes #13734

Unfortunately this breaks string freeze. So we have the choice for keeping the check 2.4 broken for the uncommon case of using deck 3 or 4 for Auto DJ, but with a translated error or fix the check introducing an English error message.

What do you think. Let's adjust the merge target accordingly.

### CODE REVIEW

File: src/library/autodj/autodjprocessor.cpp [hunk 1]
```diff
@@ -398,14 +398,20 @@ AutoDJProcessor::AutoDJError AutoDJProcessor::toggleAutoDJ(bool enable) {
...
         }
 
-        // TODO: This is a total bandaid for making Auto DJ work with decks 3
-        // and 4.  We should design a nicer way to handle this.
-        for (int i = 2; i < m_decks.length(); ++i) {
-            if (m_decks[i] && m_decks[i]->isPlaying()) {
+        // TODO: This is a total bandaid for making Auto DJ work with four decks.
+        // We should design a nicer way to handle this.
+        for (const auto& pDeck : std::as_const(m_decks)) {
+            if (pDeck == pLeftDeck) {
```
Comment (Swiftb0y) [inline src/library/autodj/autodjprocessor.cpp:404]: this omits the null check that was previously there. Can we add a `DEBUG_ASSERT` just in case?

Comment (Swiftb0y) [inline src/library/autodj/autodjprocessor.cpp:404]: this omits the null check that was previously there. Can we add a `DEBUG_ASSERT` just in case?

Comment (daschuer) [reply src/library/autodj/autodjprocessor.cpp]: Done

**Review by Swiftb0y — APPROVED** [review]
LGTM. Thank you. I think violating string freeze is acceptable here.

**Comment by daschuer** [comment]
Merge? Can one take care in @Swiftb0y's absence?

**Comment by ronso0** [comment]
Merging this now, thanks @daschuer and @Swiftb0y !

**Pull request merged**